src/test/java/com/springboot/star_wars_character_popularity/app/service/CharacterServiceUnitTest.java

```diff
@@ -90,21 +91,43 @@ void shouldReturnSavedCharacterById(){
         //when
         CharacterModel actualModel = characterService.getCharacterById(1);
 
-        
         //then
         assertEquals(actualModel, characterModel);
     }
 
     @Test
-    void shouldReturnUpdatedCharacter(){
+    void shouldThrowResourceNotFoundExceptionWhenNoCharacterAvailable(){
         //given
+        long id = 1;
 
+        when(this.characterRepositoryMock.findById(id)).thenThrow(ResourceNotFoundException.class);
 
         //when
-
+        Executable executable = () ->  characterService.getCharacterById(1);
 
         //then
-
+        assertThrows(ResourceNotFoundException.class, executable);
     }
 
+    @Test
+    void shouldReturnUpdatedCharacter(){
+        //given
+        CharacterModel characterModel = new CharacterModel("Jedi", "Naboo");
+
+        long id = 1;
+
+        Optional<CharacterModel> characterOptional = Optional.of(new CharacterModel("Jedi", "Tatooine"));
+        //CharacterModel characterOptional = new CharacterModel("Jedi", "Tatooine");
+
+        when(this.characterRepositoryMock.findById(id)).thenReturn(characterOptional);
+
+        //characterOptional.
+
+        //when
+        CharacterModel actualModel = characterService.updateCharacter(characterModel, id);
+
+        //then
+        assertEquals("Jedi", actualModel.getName());
```

Comment:
I think that this checks are unnecessary, because it's always gonna be true. (you hardcoded what you want that method to return)
Maybe it's better to use verify() method to check if the actual method is called.

Reply:
Ok. Added verify() methods.